### PR TITLE
Player object

### PR DESCRIPTION
**Pull request by n2oneProgrammer**

### CODE REVIEW

File: client/src/scripts/SceneInitializator.js [hunk 1]
```diff
@@ -13,6 +13,23 @@ export class SceneInitializator {
     async initDisplay(displayElement) {
         //Initialization Scene
         this.scene = new THREE.Scene();
+        this.renderer = new THREE.WebGLRenderer({
```
Comment (jaanonim) [inline client/src/scripts/SceneInitializator.js:16]: this code is duplicate it's move to camera class

File: client/src/scripts/Socket.js [hunk 2]
```diff
@@ -10,12 +11,14 @@ export default class Socket {
 
     constructor(room) {
         if (Socket.instance != null) return null;
+        console.log("ROOM NUMBER", room);
```
Comment (jaanonim) [inline client/src/scripts/Socket.js:14]: debug

File: client/src/scripts/utilities/Random.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
 import * as THREE from "three";
 
+const defaultCharset = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890";
```
Comment (jaanonim) [inline client/src/scripts/utilities/Random.js:3]: maybe use only caplital letters for easye copy

File: server/src/classes/Player.js [hunk 4]
```diff
@@ -1,11 +1,40 @@
 module.exports = class Player {
-    constructor(room, name, socket) {
+    constructor(room, nick, socket) {
         this.room = room;
-        this.name = name;
+        this.nick = nick;
+        this.isConnect = true;
+        this.supplies = {
```
Comment (jaanonim) [inline server/src/classes/Player.js:6]: TODO: get this form server

File: server/src/classes/Room.js [hunk 5]
```diff
@@ -41,18 +41,32 @@ module.exports = class Room {
...
+            this.redPlayer.socket = playerSocket;
+            this.redPlayer.isConnect = true;
+            return "RED";
+        }
+
         return null;
     }
 
     canStartGame() {
+        console.log("can start ", this.bluePlayer != null, this.redPlayer != null);
```
Comment (jaanonim) [inline server/src/classes/Room.js:69]: debug

File: server/src/classes/Room.js [hunk 6]
```diff
@@ -76,8 +90,11 @@ module.exports = class Room {
 
     placeFigure(player, figure) {
         //TODO: make checking if this move is OK
+        let playerObj = this.redPlayer.socket.id === player ? this.redPlayer : this.bluePlayer;
         figure.who = this.redPlayer.socket.id === player ? "RED" : "BLUE";
         let obj = this.figures.getFigure(figure.figureId, figure.figureType);
+        this.figures.supplyOperations(obj, playerObj, figure.figureType);
+        console.log(playerObj);
```
Comment (jaanonim) [inline server/src/classes/Room.js:97]: debug

File: server/src/sockets.js [hunk 7]
```diff
@@ -59,7 +62,10 @@ module.exports = {
             });
 
             socket.on("disconnecting", () => {
-                room.disconnectPlayer(socket.id);
+                if (!room.disconnectPlayer(socket.id)) {
+                    rooms = rooms.filter(r => r.name !== room.name)
+                }
+                console.log(rooms.length);
```
Comment (jaanonim) [inline server/src/sockets.js:68]: debug

**Review by jaanonim — CHANGES_REQUESTED** [review]

File: client/src/scripts/Socket.js (hunk 2 above)
Comment (n2oneProgrammer) [reply client/src/scripts/Socket.js:14]: its maybe helpfull later

**Review by jaanonim — APPROVED** [review]

**Pull request merged**